src/main/kotlin/widgets/Chip.kt

```diff
@@ -0,0 +1,77 @@
+package widgets
+
+import javafx.scene.control.Button
+import javafx.scene.control.Label
+import javafx.scene.effect.DropShadow
+import javafx.scene.input.MouseEvent
+import javafx.scene.layout.HBox
+import javafx.scene.layout.StackPane
+import javafx.scene.paint.Color
+import javafx.scene.shape.Rectangle
+import tornadofx.*
+
+/**
+ * A language chip is a class used to store the data of the chip for
+ * easy direct access. This will allow access to the stackpane, and elements
+ * in the stackpane without searching through the children on the parent of the
+ * stackpane for what we want to access.
+ */
+
+class Chip(val labelText : String,
+           fillColor : Color,
+           textColor : Color,
```

Comment:
Seems to me that colors would come through stylesheets, or type safe builders?

Reply:
The problem with this is that the color is custom to the function of the comboBox (either target or source); because of that, it seems as though we either need to pass in the color or hard-code the color in depending on the function. Can you think of any other solution? We've thought long and hard about how to determine color and can't seem to find anything better... :(

Reply:
They way the events happen, we actually don't need to set the colors inside the chip widget. Therefore, these parameters have been removed.